linker/Mono.Linker.Steps/ResolveFromAssemblyStep.cs

```diff
@@ -67,7 +67,7 @@ static void SetAction (LinkContext context, AssemblyDefinition assembly, Assembl
 		{
 			TryReadSymbols (context, assembly);
 
-			context.Annotations.SetAction (assembly, action);
+			context.SetAction (assembly, action);
 		}
```

Comment:
With this change the call to `SetAction (context, assembly, AssemblyAction.Copy);` from ProcessLibrary will not always set the action to AssemblyAction.Copy for core assemblies, which seems wrong.